docs/features/auth/certificate-authentication.md

```diff
@@ -0,0 +1,99 @@
+This filter will then add authentication to all endpoints via one or many certificate properties configurable on the filter itself.
+
+### Usage
+
+The authentication requires a service dependency to be registered with the services container of the <span>ASP.NET</span> request pipeline, which can be one of the following:
+- `ICachedSecretProvider` or `ISecretProvider`: built-in or you implementation of the secret provider
+- `Configuration`: key/value pairs in the configuration of the <span>ASP.NET</span> application
+- `IX509ValidationLocation`: custom implementation that implements the validation location interface
+
+This registration of the service is typically done in the `ConfigureServices` method of the `Startup` class.
+
+Each certificate property that should be validated can use a different service dependency. 
+This mapping of what service which property uses, is defined in an `CertificateAuthenticationValidator` instance.
+
+Once this is done, the `CertificateAuthenticationFilter` can be added to the filters that will be applied to all actions:
+
+```csharp
+public void ConfigureServices(IServiceCollections services)
+{
+    services.AddScoped<ICachedSecretProvider(serviceProvider => new MyCachedSecretProvider());
+
+    services.AddScoped<CertificateAuthenticationValidator>(
+        serviceProvider => new CertificateAuthenticationValidator()
+            .WithRequirementLocation(X509ValidationRequirement.SubjectName, X509ValidationLocation.SecretProvider));
+
+    services.AddMvc(
+        options => options.Filters.Add(
+            new CertificateAuthenticationFilter(
+                X509ValidationRequirement.SubjectName,
+                "key-to-certificate-subject-name"
+            )));
+}
+```
+
+## Enforce certificate authentication per controller or operation
+
+### Introduction
+
+The `CertificateAuthenticationAttribute` can be added on both controller- and operation level in an <span>ASP.NET</span> Core application.
+This certificate authentication will then be applied to the endpoint(s) that are decorated with the `CertificateAuthenticationAttribute`.
+
+### Usage
+
+The authentication requires a service dependency to be registered with the services container of the <span>ASP.NET</span> request pipeline, which can be one of the following:
+- `ICachedSecretProvider` or `ISecretProvider`: built-in or you implementation of the secret provider
+- `Configuration`: key/value pairs in the configuration of the <span>ASP.NET</span> application
+- `IX509ValidationLocation`: custom implementation that implements the validation location interface
+
+This registration of the service is typically done in the `ConfigureServices` method of the `Startup` class:
+
+```csharp
+public void ConfigureServices(IServiceCollections services)
+{
+    services.AddScoped<ICachedSecretProvider(serviceProvider => new MyCachedSecretProvider());
```

Comment:
Here you would have to do the same setup as mentioned above and then in the controller itself you can decorate it simply with `[CertificateAuthentication)]`.

If people want more flexibility *later on*, we can track this in the backlog to create some sort of profiles, but this is too far down the road for now.

Reply:
Outdated...